# TAREA 3
Maria Fernanda Gómez Flores    
Lunes 14 de septiembre 2026

Asume que eres el Portfolio Manager de Banamex y deseas comparar el desempeño de una estrategia de inversión activa con una pasiva (Benchmark), realizando un análisis básico del riesgo y rendimiento de ambas estrategias. Sigue los pasos para evaluarlas:

1.- Tu estrategia de inversión activa contiene las sigtes. acciones con ponderaciones equitativas JP Morgan Chase, Visa, Goldman Sachs Group Inc, Progressive Corp, para tu portafolio:

Descarga los precios de cierre ajustado desde el 2018-01-01 hasta el 2026-09-07 .
Calcula las métricas de riesgo y rendimiento anuales, coeficiente de variación, ratio de Sharpe y la matriz de correlación. (20%)
Interpreta la matriz de correlación. (10%)
 

2.- Ahora, compara tu estrategia de inversión activa contra una pasiva:

En base a tus acciones y su sector del GICS busca un índice que represente el sector de tus activos, comenta el nombre, ticker y los constituyentes de este. Descarga los precios del índice Yahoo Finance. (10%)
Para el índice, calcula las métricas de riesgo y rendimiento anuales, además del ratio de Sharpe. (20%)
 

3. -A manera de conclusión, en función de las métricas calculadas y en base a los dos Radios de Sharpe calculados para cada estrategia, responde:

¿En que estrategia invertirías y por qué?, ¿Habría una forma de mejorar la estrategia activa? es decir, ¿Qué podrías cambiar en tu portafolio para obtener mejores resultados? (20%)
Calcula el rendimiento geométrico y armónico para la estrategia pasiva y activa, discute en que casos sería conveniente utilizar uno u otro (20%)
Nota: Asume una tasa libre de riesgo del 4% anual.
Entrega tu tarea en un html descargado de Jupyter Notebook.

In [1]:
import pandas as pd
import numpy as np
import yfinance as yf
import matplotlib.pyplot as plt

In [2]:
tickers=['JPM','V','GS','PGR','^SP500-40']
prices=yf.download(tickers,start='2018-01-01',end='2026-09-07')['Close']
rets=(prices/prices.shift()-1)
rets
rf=0.04

/var/folders/x9/60hcc2ln62d3r3xyv90d3wl40000gn/T/ipykernel_83107/597710148.py:2: FutureWarning: YF.download() has changed argument auto_adjust default to True
  prices=yf.download(tickers,start='2018-01-01',end='2026-09-07')['Close']
[*********************100%***********************]  5 of 5 completed


## 1. ESTRATEGIA ACTIVA

In [3]:
retsA=rets.drop(['^SP500-40'],axis=1).dropna()
w_portA=np.array([.25,.25,.25,.25])
cov=retsA.cov()


In [4]:
daily_retsA=retsA.mean()
mu=daily_retsA*252

E_R_portA=sum(mu*w_portA)
E_R_portA*100

21.007784968474912

In [5]:

daily_varianceA=w_portA.T@cov@w_portA
annual_varianceA=daily_varianceA*252
annual_varianceA

np.float64(0.04964934050814852)

In [6]:
daily_volatilityA=np.sqrt(daily_varianceA)
annual_volatilityA=np.sqrt(annual_varianceA)

annual_volatilityA*100

np.float64(22.282131968945098)

In [7]:
Sharpe_RatioA= (E_R_portA-rf)/annual_volatilityA
Sharpe_RatioA

np.float64(0.7632925337745456)

In [8]:
cvA= annual_volatilityA/E_R_portA
cvA

np.float64(1.0606607028005344)

In [9]:
corr=retsA.corr()
corr

Ticker,GS,JPM,PGR,V
Ticker,,,,
GS,1.000000,0.804611,0.304207,0.535723
JPM,0.804611,1.000000,0.392582,0.579559
PGR,0.304207,0.392582,1.000000,0.413230
V,0.535723,0.579559,0.413230,1.000000


La correlación más alta se encuentra entre GS y JPM con 0.81, por lo que estos activos tienden a moverse de manera similar y aportan menos diversificación entre sí. La correlación más baja es entre GS y PGR con 0.32, por lo que estos tienen una relación menor y ayudan más a diversificar el portafolio.

In [10]:
rets_portA = retsA @ w_portA

geometric_returnA = (((rets_portA+1).prod())**(1/len(rets_portA))-1)*252
geometric_returnA*100


np.float64(18.526431655175468)

In [11]:
harmonic_returnA = (len(rets_portA)/((1/(rets_portA+1)).sum())-1)*252
harmonic_returnA*100

np.float64(16.0395818109202)

## 2. ESTRATEGIA PASIVA

Para la estrategia pasiva usé como benchmark el S&P 500 Financials con ticker ^SP500-40, porque representa al sector financiero y sirve como comparación para mi portafolio activo. Este índice incluye empresas financieras importantes del S&P 500, como JPMorgan Chase, Visa, Goldman Sachs, Bank of America, Mastercard, Berkshire Hathaway y Wells Fargo.

In [12]:
retsP=rets[['^SP500-40']].dropna()


In [13]:
daily_retsP=retsP.mean()
muP=daily_retsP*252
muP*100

Ticker
^SP500-40    10.777523
dtype: float64

In [14]:
daily_varianceP=retsP.var()
annual_varianceP=daily_varianceP*252
annual_varianceP

Ticker
^SP500-40    0.054575
dtype: float64

In [15]:
daily_volatilityP=np.sqrt(daily_varianceP)
annual_volatilityP=np.sqrt(annual_varianceP)
annual_volatilityP*100

Ticker
^SP500-40    23.361256
dtype: float64

In [16]:
Sharpe_RatioP= (muP-rf)/annual_volatilityP
Sharpe_RatioP

Ticker
^SP500-40    0.290118
dtype: float64

In [17]:
cvP= annual_volatilityP/muP
cvP

Ticker
^SP500-40    2.16759
dtype: float64

In [18]:
geometric_returnP=(((retsP+1).prod())**(1/len(retsP))-1)*252
geometric_returnP*100


Ticker
^SP500-40    8.040202
dtype: float64

In [19]:
harmonic_returnP=(len(retsP)/((1/(retsP+1)).sum())-1)*252
harmonic_returnP*100

Ticker
^SP500-40    5.287073
dtype: float64

## 3. CONCLUSIONES

In [20]:
E_R_portA*100,muP*100

(21.007784968474912,
 Ticker
 ^SP500-40    10.777523
 dtype: float64)

In [21]:
annual_volatilityA*100,annual_volatilityP*100

(np.float64(22.282131968945098),
 Ticker
 ^SP500-40    23.361256
 dtype: float64)

In [22]:
Sharpe_RatioA,Sharpe_RatioP

(np.float64(0.7632925337745456),
 Ticker
 ^SP500-40    0.290118
 dtype: float64)

In [23]:
cvA,cvP

(np.float64(1.0606607028005344),
 Ticker
 ^SP500-40    2.16759
 dtype: float64)

In [24]:
geometric_returnA*100,geometric_returnP*100

(np.float64(18.526431655175468),
 Ticker
 ^SP500-40    8.040202
 dtype: float64)

In [25]:
harmonic_returnA*100,harmonic_returnP*100

(np.float64(16.0395818109202),
 Ticker
 ^SP500-40    5.287073
 dtype: float64)

Yo invertiría en la estrategia activa, porque tuvo un rendimiento anual de 21.01%, mientras que la pasiva tuvo 10.78%. Aunque las dos tienen un nivel de riesgo parecido, la activa tuvo una volatilidad un poco menor, de 22.28%, contra 23.36% de la pasiva. También el Sharpe de la activa fue mayor, 0.7633 contra 0.2901, lo que significa que tuvo un mejor rendimiento tomando en cuenta el riesgo. Lo mismo pasa con el CV, ya que la activa tuvo 1.061 y la pasiva 2.168, y en este caso entre más bajo sea el CV es mejor.

Creo que una forma de mejorar la estrategia activa sería cambiar o reducir la ponderación de alguno de los activos que tienen una correlación muy alta. Por ejemplo, GS y JPM tienen una correlación de aproximadamente 0.80, entonces se mueven bastante parecido. Se podría cambiar alguno por un activo que tenga una correlación más baja con los demás para tener una mejor diversificación y tratar de reducir el riesgo.

Por último, en la estrategia activa el rendimiento geométrico fue de 18.53% y el armónico de 16.04%, mientras que en la pasiva fueron 8.04% y 5.29% respectivamente. El geométrico sería más conveniente para ver cómo ha crecido realmente una inversión a través del tiempo porque toma en cuenta el efecto compuesto. El armónico le da más peso a los rendimientos bajos, entonces puede servir cuando queremos tomar más en cuenta los periodos donde la inversión tuvo un menor desempeño. En general, con los resultados obtenidos considero que la estrategia activa tuvo un mejor desempeño que la pasiva.

